util/xrh_header.go

```diff
@@ -0,0 +1,45 @@
+package util
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"github.com/RedHatInsights/sources-api-go/kafka"
+	"github.com/redhatinsights/platform-go-middlewares/identity"
+)
+
+func ParseXRHIDHeader(inputIdentity string) (XRHIdentity *identity.XRHID, err error) {
+	decodedIdentity, err := base64.StdEncoding.DecodeString(inputIdentity)
+	if err != nil {
+		return nil, fmt.Errorf("error decoding Identity: %v", err)
+	}
+
+	err = json.Unmarshal(decodedIdentity, &XRHIdentity)
+	if err != nil {
+		return nil, fmt.Errorf("x-rh-identity header does not contain valid JSON")
+	}
+
+	return XRHIdentity, nil
+}
+
+func AccountNumberFrom(headers []kafka.Header) (string, error) {
+	XRHAccountNumberKey := "x-rh-sources-account-number"
+	XRHIdentityKey := "x-rh-identity"
```

Comment:
I would put these on the package level as a couple of vars, so they don't get instantiated every time